mountpoint-s3/src/fstab.rs

```diff
@@ -0,0 +1,252 @@
+use anyhow::anyhow;
+use clap::error::ErrorKind;
+use clap::{Command, CommandFactory, Parser};
+use mountpoint_s3_fs::cli::CliArgs;
+use std::env;
+
+#[derive(Parser, Debug)]
+#[clap(name = "mount-s3", disable_help_flag = true)]
+pub(crate) struct FsTabCliArgs {
+    bucket_name: String,
+    #[clap(value_name = "DIRECTORY")]
+    mount_point: String,
+    // Needs to be explicit `std::vec::Vec` because of https://github.com/clap-rs/clap/issues/4808
+    #[clap(short = 'o', value_parser = split_commas)]
+    options: std::vec::Vec<String>,
+}
+
+impl TryFrom<FsTabCliArgs> for CliArgs {
+    type Error = clap::Error;
+
+    fn try_from(fstab_cli_args: FsTabCliArgs) -> Result<Self, Self::Error> {
+        let cli_arg_list = fstab_cli_args.into_cli_arg_list()?;
+
+        let mut cli_args = CliArgs::try_parse_from(cli_arg_list)?;
+        cli_args.foreground = true;
+        Ok(cli_args)
+    }
+}
+
+impl FsTabCliArgs {
+    /// Parse the args we've been given into an iterator of options to pass to the main CliArgs parser
+    /// Filters out arguments that aren't meant for Mountpoint, and add `--` to any argument that's not prefixed with `-`.
+    fn into_cli_arg_list(self) -> Result<Vec<String>, clap::Error> {
+        Self::validate_options(&self.options)?;
+
+        let cli_arg_list = [
+            env::args().nth(0).unwrap_or("mount-s3".to_string()),
+            self.bucket_name,
+            self.mount_point,
+        ]
+        .into_iter()
+        .chain(
+            self.options
+                .into_iter()
+                .filter(|option| !Self::filter_option(option))
```

Comment:
nit: if you invert the meaning of `filter_option` to be in line with the `filter` predicate, you should be able to do:
```suggestion
                .filter(Self::filter_option)
```